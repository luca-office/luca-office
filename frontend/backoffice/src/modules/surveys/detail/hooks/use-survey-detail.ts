import {pick} from "lodash-es"
import {useState} from "react"
import {useForm, UseFormMethods} from "react-hook-form"
import {useDispatch} from "react-redux"
import {SurveyUpdate} from "shared/graphql/generated/globalTypes"
import {
  ProjectProps,
  SurveyProps,
  useProject,
  useRatings,
  useSurvey,
  useSurveyResultsOverview,
  useSurveyUserAccounts,
  useUpdateSurvey
} from "shared/graphql/hooks"
import {CompletedParticipantCount, SurveyResultsOverview} from "shared/models"
import {Route as SharedRoute} from "shared/routes"
import {find, getCompletedParticipantsCount, isDefined, Option} from "shared/utils"
import {isSurveyFinished} from "shared/utils/survey-finished"
import {SurveyTiming} from "../../../../enums"
import {UserAccount} from "../../../../models"
import {navigateToRouteAction} from "../../../../redux/actions/navigation-action"
import {Route} from "../../../../routes"
import {getManualSurveyTimingStatus, getSurveyTimingStatus} from "../../../../utils"
import {surveyPollingRate} from "../../../monitoring/config/config"
import {isManualSurvey} from "../../utils/common"

export interface SurveyDetailForm {
  readonly title: string
  readonly description: string
}

export interface UseSurveyDetailHook extends Pick<SurveyProps, "survey">, Pick<ProjectProps, "project"> {
  readonly author: Option<UserAccount>
  readonly dataLoading: boolean
  readonly formMethods: UseFormMethods<SurveyDetailForm>
  readonly inviteAttendeesModalVisible: boolean
  readonly isEditable: boolean
  readonly isInvitationEnabled: boolean
  readonly navigateToDashboard: () => void
  readonly navigateToProjectDetail: () => void
  readonly navigateToSurveyUpdate: () => void
  readonly toggleInviteAttendeesModal: () => void
  readonly navigateToRatingOverview: () => void
  readonly navigateToReportingOverview: () => void
  readonly updateInProgress: boolean
  readonly updateSurvey: (id: string, update: SurveyDetailForm) => void
  readonly surveyResultsOverview: Option<SurveyResultsOverview>
  readonly isRatingFinalized: boolean
  readonly raters: UserAccount[]
  readonly completedParticipantsCount: CompletedParticipantCount
}

export const useSurveyDetail = (projectId: UUID, surveyId: UUID): UseSurveyDetailHook => {
  const dispatch = useDispatch()
  const [showInviteAttendeesModal, setShowInviteAttendeesModal] = useState(false)

  const {updateSurvey, updateSurveyLoading} = useUpdateSurvey()
  const {survey: surveyOption, surveyLoading} = useSurvey(surveyId)
  const {project: projectOption, projectLoading} = useProject(projectId)
  const {ratings: allRatings, ratingsLoading} = useRatings(surveyId)
  const {surveyResultsOverview, surveyResultsOverviewLoading} = useSurveyResultsOverview(
    surveyId,
    false,
    surveyOption.exists(survey => isSurveyFinished(survey)) ? undefined : surveyPollingRate
  )
  const {surveyUserAccounts: raters} = useSurveyUserAccounts(surveyId)

  const isManualSurveyExecutionType = isManualSurvey(surveyOption.map(survey => survey.executionType))

  const isRatingFinalized = find(({isFinalScore}) => isFinalScore, allRatings).exists(({finalizedAt}) =>
    isDefined(finalizedAt)
  )

  const formMethods = useForm<SurveyDetailForm>()

  const {isEditable, isInvitationEnabled} = surveyOption
    .map(survey => {
      const surveyTimingStatus = isManualSurveyExecutionType
        ? getManualSurveyTimingStatus(survey)
        : getSurveyTimingStatus(survey.startsAt, survey.endsAt)
      return {
        isEditable: surveyTimingStatus === SurveyTiming.Future && !survey.isCompleted,
        isInvitationEnabled:
          (surveyTimingStatus === SurveyTiming.Present || surveyTimingStatus === SurveyTiming.Future) &&
          !survey.isCompleted
      }
    })
    .getOrElse({isEditable: false, isInvitationEnabled: false})

  const navigateToProjectDetail = () => dispatch(navigateToRouteAction(Route.ProjectDetail, {id: projectId}))
  const navigateToSurveyUpdate = () => dispatch(navigateToRouteAction(Route.SurveyEdit, {id: projectId, surveyId}))
  const navigateToDashboard = () =>
    dispatch(
      navigateToRouteAction(SharedRoute.SurveyMonitoring, {
        projectId,
        surveyId
      })
    )
  const navigateToRatingOverview = () => dispatch(navigateToRouteAction(Route.SurveyScoring, {projectId, surveyId}))
  const navigateToReportingOverview = () =>
    dispatch(navigateToRouteAction(Route.SurveyReporting, {projectId, surveyId}))

  const handleUpdate = (id: UUID, update: SurveyDetailForm) => {
    updateSurvey(id, ({
      ...surveyOption
        .map(survey =>
          pick(survey, "startsAt", "endsAt", "authenticationType", "isOpenParticipationEnabled", "executionType")
        )
        .orUndefined(),
      ...update
    } as unknown) as SurveyUpdate).then(response =>
      response.forEach(entity =>
        dispatch(navigateToRouteAction(Route.SurveyDetail, {id: projectId, surveyId: entity.id}))
      )
    )
  }

  const completedParticipantsCount = surveyResultsOverview.map(getCompletedParticipantsCount).getOrElse({
    numCompletedParticipants: 0,
    totalParticipants: 0
  })

  return {
    isEditable,
    isInvitationEnabled,
    author: projectOption.map(project => project.author),
    dataLoading: projectLoading || ratingsLoading || surveyLoading || surveyResultsOverviewLoading,
    formMethods,
    navigateToDashboard,
    navigateToProjectDetail,
    navigateToSurveyUpdate,
    navigateToRatingOverview,
    navigateToReportingOverview,
    survey: surveyOption,
    project: projectOption,
    updateInProgress: updateSurveyLoading,
    updateSurvey: handleUpdate,
    inviteAttendeesModalVisible: showInviteAttendeesModal,
    toggleInviteAttendeesModal: () => setShowInviteAttendeesModal(!showInviteAttendeesModal),
    surveyResultsOverview,
    isRatingFinalized,
    raters,
    completedParticipantsCount
  }
}
