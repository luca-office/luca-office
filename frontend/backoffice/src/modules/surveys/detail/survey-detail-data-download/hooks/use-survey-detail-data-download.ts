import * as React from "react"

export interface UseSurveyDetailDataDownloadHook {
  readonly isConfirmModalVisible: boolean
  readonly showConfirmModal: () => void
  readonly hideConfirmModal: () => void
}

export const useSurveyDetailDataDownload = (): UseSurveyDetailDataDownloadHook => {
  const [isConfirmModalVisible, setIsConfirmModalVisible] = React.useState(false)

  const showConfirmModal = () => setIsConfirmModalVisible(true)
  const hideConfirmModal = () => setIsConfirmModalVisible(false)

  return {isConfirmModalVisible, showConfirmModal, hideConfirmModal}
}
