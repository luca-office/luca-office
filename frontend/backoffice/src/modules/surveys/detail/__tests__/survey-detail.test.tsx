import {MockedProvider} from "@apollo/client/testing"
import {mount} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {
  Card,
  CardContent,
  CardFooter,
  Content,
  ContentBottomActionbar,
  ContentLoadingIndicator,
  ContentMissingIndicator,
  DeleteOrArchiveEntityButton,
  DetailViewHeader,
  LoadingIndicator,
  ReadonlyActionField
} from "shared/components"
import {projectsMock, surveysMock, userAccountsMock} from "shared/graphql/__mocks__"
import {Option} from "shared/utils"
import {mockedFormMethods} from "sharedTests/utils/form-methods-mock"
import {OverlayEditField} from "../../../../components"
import {surveyResultsOverviewMock} from "../../../../graphql/__mocks__"
import {DashboardActionFooter, SurveyProgress} from "../../../dashboard/common"
import * as projectDetailHook from "../hooks/use-survey-detail"
import {UseSurveyDetailHook} from "../hooks/use-survey-detail"
import {InviteSurveyAttendeesModalContainer} from "../invite-attendees/invite-survey-attendees-modal-container"
import {SurveyDetail} from "../survey-detail"
import {SurveyDetailDataDownload} from "../survey-detail-data-download/survey-detail-data-download"
import {SurveyDetailSettings} from "../survey-detail-settings/survey-detail-settings"

const survey = surveysMock[0]
const project = projectsMock[0]

const hookValuesDefault: UseSurveyDetailHook = {
  survey: Option.of(survey),
  project: Option.of(project),
  author: Option.of(project.author),
  dataLoading: false,
  formMethods: mockedFormMethods,
  inviteAttendeesModalVisible: false,
  isEditable: true,
  navigateToDashboard: jest.fn(),
  navigateToProjectDetail: jest.fn(),
  navigateToReportingOverview: jest.fn(),
  navigateToSurveyUpdate: jest.fn(),
  toggleInviteAttendeesModal: jest.fn(),
  navigateToRatingOverview: jest.fn(),
  updateInProgress: false,
  updateSurvey: jest.fn(),
  surveyResultsOverview: Option.of(surveyResultsOverviewMock),
  isRatingFinalized: false,
  isInvitationEnabled: false,
  raters: userAccountsMock,
  completedParticipantsCount: {
    numCompletedParticipants: 3,
    totalParticipants: 6
  }
}

const stateSpy = jest.spyOn(projectDetailHook, "useSurveyDetail")
const getComponent = () => (
  <MockedProvider>
    <SurveyDetail projectId={project.id} surveyId={survey.id} />
  </MockedProvider>
)

describe("survey-detail", () => {
  it("renders correctly", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)
    const component = create(getComponent())

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure (with project)", async () => {
    stateSpy.mockReturnValue(hookValuesDefault)

    const component = mount(getComponent())
    expect(component.find(Content)).toHaveLength(1)
    expect(component.find(ContentBottomActionbar)).toHaveLength(1)
    expect(component.find(DashboardActionFooter)).toHaveLength(1)
    expect(component.find(DetailViewHeader)).toHaveLength(1)
    expect(component.find(DetailViewHeader).find(DeleteOrArchiveEntityButton)).toHaveLength(1)
    expect(component.find(Card)).toHaveLength(6)
    expect(component.find(Card).find(CardContent)).toHaveLength(6)
    expect(component.find(Card).find(CardFooter)).toHaveLength(3)
    expect(component.find(Card).find(SurveyDetailSettings)).toHaveLength(1)
    expect(component.find(Card).find(SurveyDetailDataDownload)).toHaveLength(1)
    expect(component.find(Card).find(SurveyProgress)).toHaveLength(1)
    expect(component.find(OverlayEditField)).toHaveLength(1)
    expect(component.find(ReadonlyActionField)).toHaveLength(3)
  })
  it("has correct structure (loading)", async () => {
    stateSpy.mockReturnValue({...hookValuesDefault, survey: Option.none(), dataLoading: true})

    const component = mount(getComponent())
    expect(component.find(ContentLoadingIndicator)).toHaveLength(1)
    expect(component.find(LoadingIndicator)).toHaveLength(1)
  })
  it("has correct structure (project not found)", async () => {
    stateSpy.mockReturnValue({...hookValuesDefault, survey: Option.none()})

    const component = mount(getComponent())
    expect(component.find(ContentMissingIndicator)).toHaveLength(1)
  })
  it("has correct structure (inviteAttendeesModalVisible)", async () => {
    stateSpy.mockReturnValue({...hookValuesDefault, inviteAttendeesModalVisible: true})

    const component = mount(getComponent())
    expect(component.find(Content)).toHaveLength(1)
    expect(component.find(ContentBottomActionbar)).toHaveLength(1)
    expect(component.find(DashboardActionFooter)).toHaveLength(1)
    expect(component.find(DetailViewHeader)).toHaveLength(1)
    expect(component.find(DetailViewHeader).find(DeleteOrArchiveEntityButton)).toHaveLength(1)
    expect(component.find(Card)).toHaveLength(6)
    expect(component.find(Card).find(CardContent)).toHaveLength(6)
    expect(component.find(Card).find(CardFooter)).toHaveLength(3)
    expect(component.find(Card).find(SurveyDetailSettings)).toHaveLength(1)
    expect(component.find(Card).find(SurveyDetailDataDownload)).toHaveLength(1)
    expect(component.find(Card).find(SurveyProgress)).toHaveLength(1)
    expect(component.find(OverlayEditField)).toHaveLength(1)
    expect(component.find(ReadonlyActionField)).toHaveLength(3)
    expect(component.find(InviteSurveyAttendeesModalContainer)).toHaveLength(1)
  })
})
