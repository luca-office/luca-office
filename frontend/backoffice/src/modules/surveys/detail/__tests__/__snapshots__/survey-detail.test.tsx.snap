// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`survey-detail renders correctly 1`] = `
<div
  className="content css-r3hgbg-Content"
>
  <div
    className="css-9pumgg-DetailViewHeader"
  >
    <button
      className="css-rmd9d0-Button"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="css-95j2hm-Button"
      >
        <div
          className="css-1lrzij5-Icon"
          dangerouslySetInnerHTML={
            {
              "__html": {},
            }
          }
        />
        <div
          className="css-u7npei-DetailViewHeader"
        >
          projects__survey_details_header_navigate_back_label
        </div>
      </div>
    </button>
    <div
      className="css-tqo6e0-DetailViewHeader"
    >
      projects__survey_details_header_label
    </div>
    <div
      className="css-16tq2dx-DetailViewHeader"
    >
      <div
        className="css-17bom2c-OrlyButton"
      >
        <button
          className="css-hnekdl-Button"
          disabled={true}
          onClick={[Function]}
        >
          <div
            className="css-95j2hm-Button"
          >
            <div
              className="css-mj8vo1-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="content-container css-6a1htv-Content"
  >
    <div
      className="css-63bajj-Card"
    >
      <div
        className="card-content css-1n9n9nm-CardContent"
      >
        <div
          className="css-1a50cz1-SurveyDetail"
        >
          <div>
            <div
              className="css-lz4b3k-Label"
            >
              title
              <div
                className="css-nhflpf-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
            </div>
            <div
              className="css-1yd7xcl-InlineEditableHeader"
            >
              <div>
                <form
                  className="css-1c5ej61-InlineEditableHeader"
                  onBlur={[Function]}
                  onSubmit={[Function]}
                >
                  <div
                    className="css-pl4s5v-InlineEditableHeader"
                    onClick={[Function]}
                  >
                    <div
                      className="css-1iwo50k"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                    >
                      <div
                        className="css-nrul0i"
                      >
                        <input
                          className="css-q6vozw"
                          defaultValue="my test scenario"
                          disabled={false}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          readOnly={true}
                          type="text"
                        />
                      </div>
                    </div>
                    <div
                      className="hover-icon css-16vb3lw-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </form>
              </div>
            </div>
          </div>
          <div
            className="css-u6fbpf-Heading"
          >
            dashboard__title_time_ended
          </div>
        </div>
        <div
          className="css-lz4b3k-Label"
        >
          description
          <div
            className="css-nhflpf-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
        <div
          className="overlay-edit-field css-0"
        >
          <div
            className="css-1kjt8hf-ReadonlyActionField"
            data-testid="readonly-action-field"
            onClick={[Function]}
          >
            <div
              className="css-1bkess4-ReadonlyActionField"
            />
            <div
              className="content-wrapper css-376q3o-ReadonlyActionField"
            >
              <div
                className="content css-ci7wec-ReadonlyActionField"
              >
                <div
                  className="css-1h2eymq-InlineEditableTextarea"
                  onClick={[Function]}
                >
                  <div
                    className="textarea css-1n5bn12-InlineEditableTextarea"
                    contentEditable={false}
                    suppressContentEditableWarning={true}
                  >
                    description
                  </div>
                  <div
                    className="hover-icon css-1pndqgp-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-14rjq1s-SurveyDetail"
        >
          <div
            className="css-jkinj2-ReadonlyActionField"
            data-testid="readonly-action-field"
          >
            <div
              className="css-1bkess4-ReadonlyActionField"
            >
              <div
                className="css-47wnmy-ReadonlyActionField"
              >
                projects__surveys_details_participants_label
              </div>
            </div>
            <div
              className="content-wrapper css-b45n67-ReadonlyActionField"
            >
              <div
                className="content css-1hje2zc-ReadonlyActionField"
              >
                <div
                  className="css-hsuzhl-Text"
                >
                  <div
                    className="css-1lrzij5-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                  projects__surveys_details_participants_value
                </div>
              </div>
              <button
                className="readonly-action-button css-4nv627-Button"
                disabled={true}
                onClick={[Function]}
              >
                <div
                  className="css-95j2hm-Button"
                >
                  invite_button
                </div>
              </button>
            </div>
          </div>
          <div
            className="css-yle8de-ReadonlyActionField"
            data-testid="readonly-action-field"
            onClick={[MockFunction]}
          >
            <div
              className="css-1bkess4-ReadonlyActionField"
            >
              <div
                className="css-47wnmy-ReadonlyActionField"
              >
                projects__surveys_details_project_label
              </div>
            </div>
            <div
              className="content-wrapper css-b45n67-ReadonlyActionField"
            >
              <div
                className="content css-1hje2zc-ReadonlyActionField"
              >
                <div
                  className="css-1qdd775-Text"
                >
                  <span>
                    Project 1
                  </span>
                </div>
              </div>
              <button
                className="readonly-action-button css-4nv627-Button"
                disabled={false}
                onClick={[Function]}
              >
                <div
                  className="css-95j2hm-Button"
                >
                  show_button
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          className="css-1hol3sj-SurveyDetail"
        >
          <div
            className="css-148j23u-SurveyDetail"
          >
            <div
              className="css-1wutlzu-LabelledCard"
            >
              <div
                className="css-lz4b3k-Label"
              >
                projects__survey_details_settings_label
              </div>
              <div
                className="css-lblbfe-Card"
              >
                <div
                  className="card-content css-1vyr92j-CardContent"
                >
                  <div
                    className="css-1mnv1ci-SurveyDetailSettings"
                  >
                    <div
                      className="css-1ulrwv1-InformationEntry"
                    >
                      <div
                        className="css-7wq07p-InformationEntry"
                      >
                        <div
                          className="css-uoep3g-InformationEntry"
                        >
                          projects__surveys_details_participation_label
                        </div>
                      </div>
                      <div
                        className="css-plf496-InformationEntry"
                      >
                        <div
                          className="css-17zbm7y-Text"
                        >
                          projects__survey_overlay_closed_participation
                        </div>
                      </div>
                    </div>
                    <div
                      className="css-1ulrwv1-InformationEntry"
                    >
                      <div
                        className="css-7wq07p-InformationEntry"
                      >
                        <div
                          className="css-uoep3g-InformationEntry"
                        >
                          projects__surveys_progress_section_label
                        </div>
                      </div>
                      <div
                        className="css-plf496-InformationEntry"
                      >
                        <div
                          className="css-17zbm7y-Text"
                        >
                          projects__surveys_details_progress_automatic
                        </div>
                      </div>
                    </div>
                  </div>
                  <button
                    className="css-12cowym-Button"
                    disabled={false}
                    onClick={[Function]}
                  >
                    <div
                      className="css-95j2hm-Button"
                    >
                      edit_button
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              className="css-v3urfw-SurveyDetailDataDownload"
            >
              <div
                className="css-1rabetu-SurveyDetailDataDownload"
              >
                <div
                  className="css-4d08oz-Heading"
                >
                  projects__survey_details_data
                </div>
              </div>
              <div
                className="css-lblbfe-Card"
              >
                <div
                  className="card-content css-1gc675l-CardContent"
                >
                  <div
                    className="css-r9z753-SurveyDetailDataDownload"
                  >
                    <div
                      className="css-1ds7t6g-SurveyDetailDataDownload"
                    >
                      projects__survey_details_survey_data
                      <div
                        className="css-0"
                        id="tooltip_42"
                      >
                        <div
                          className="css-qnyrc1-Icon"
                          dangerouslySetInnerHTML={
                            {
                              "__html": {},
                            }
                          }
                        />
                      </div>
                      <div
                        className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                        role="tooltip"
                        style={
                          {
                            "backgroundColor": "#2e3032",
                            "zIndex": 1,
                          }
                        }
                      >
                        <div
                          className="css-12xum2z-Tooltip"
                        >
                          <div
                            className="css-1kjzr28-Tooltip"
                          >
                            <div
                              className="css-jb7kz7-Heading"
                            >
                              projects__survey_details_survey_data
                            </div>
                            <div
                              className="css-8tih5m-Text"
                            >
                              projects__survey_details_survey_data_tooltip
                            </div>
                          </div>
                        </div>
                        <div
                          className="react-tooltip-arrow styles-module_arrow__K0L3T"
                          style={{}}
                        />
                      </div>
                    </div>
                    <a
                      className="css-dc975c-SurveyDetailDataDownload"
                      href="/backend/binary/survey/18de306c-a8f8-4311-9a2f-ca5418a245ac"
                    >
                      <button
                        className="css-1cywa3f-Button"
                        disabled={false}
                        onClick={[Function]}
                      >
                        <div
                          className="css-95j2hm-Button"
                        >
                          <div
                            className="css-mj8vo1-Icon"
                            dangerouslySetInnerHTML={
                              {
                                "__html": {},
                              }
                            }
                          />
                        </div>
                      </button>
                    </a>
                  </div>
                  <div
                    className="css-r9z753-SurveyDetailDataDownload"
                  >
                    <div
                      className="css-1ds7t6g-SurveyDetailDataDownload"
                    >
                      projects__survey_details_media
                      <div
                        className="css-0"
                        id="tooltip_42"
                      >
                        <div
                          className="css-qnyrc1-Icon"
                          dangerouslySetInnerHTML={
                            {
                              "__html": {},
                            }
                          }
                        />
                      </div>
                      <div
                        className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                        role="tooltip"
                        style={
                          {
                            "backgroundColor": "#2e3032",
                            "zIndex": 1,
                          }
                        }
                      >
                        <div
                          className="css-12xum2z-Tooltip"
                        >
                          <div
                            className="css-1kjzr28-Tooltip"
                          >
                            <div
                              className="css-jb7kz7-Heading"
                            >
                              projects__survey_details_media
                            </div>
                            <div
                              className="css-8tih5m-Text"
                            >
                              projects__survey_details_media_tooltip
                            </div>
                          </div>
                        </div>
                        <div
                          className="react-tooltip-arrow styles-module_arrow__K0L3T"
                          style={{}}
                        />
                      </div>
                    </div>
                    <a
                      className="css-dc975c-SurveyDetailDataDownload"
                      href="/backend/binary/survey/18de306c-a8f8-4311-9a2f-ca5418a245ac/binaries"
                    >
                      <button
                        className="css-1cywa3f-Button"
                        disabled={false}
                        onClick={[Function]}
                      >
                        <div
                          className="css-95j2hm-Button"
                        >
                          <div
                            className="css-mj8vo1-Icon"
                            dangerouslySetInnerHTML={
                              {
                                "__html": {},
                              }
                            }
                          />
                        </div>
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="css-1d0gwig-SurveyDetail"
            >
              <div
                className="css-10550zi-Card"
                onClick={[MockFunction]}
              >
                <div
                  className="card-header css-17xm4q9-CardHeader"
                >
                  <div
                    className="css-1cvfayx-Heading"
                  >
                    dashboard__progress_project_title
                  </div>
                  <div
                    className="css-1pmfd-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
                <div
                  className="card-content css-51sp4r-CardContent"
                >
                  <div
                    className="donut-chart css-nufzg-DonutChartWithLegend"
                  >
                    <div
                      className="css-ow8vil-DonutChartWithLegend"
                    >
                      <svg
                        height="100%"
                        viewBox="0 0 100 100"
                        width="100%"
                      >
                        <path
                          d="M 50 95 A 45 45 0 1 1 50.00078539816336 94.9999999931461"
                          fill="none"
                          stroke="#d5d5d5"
                          strokeWidth={10}
                        />
                        <path
                          d="M 50 95 A 45 45 0 0 1 5 50.00000000000001"
                          fill="none"
                          stroke="#b4b4b4"
                          strokeWidth={10}
                          style={{}}
                        >
                          <title>
                            dashboard__progress_chart_invited
                          </title>
                        </path>
                        <path
                          d="M 5 50.00000000000001 A 45 45 0 0 1 49.99999999999999 5"
                          fill="none"
                          stroke="#4f82cf"
                          strokeWidth={10}
                          style={{}}
                        >
                          <title>
                            dashboard__progress_chart_active
                          </title>
                        </path>
                        <path
                          d="M 49.99999999999999 5 A 45 45 0 0 1 50.000000000000014 95"
                          fill="none"
                          stroke="#4fcfc5"
                          strokeWidth={10}
                          style={{}}
                        >
                          <title>
                            dashboard__progress_chart_completed
                          </title>
                        </path>
                      </svg>
                      <div
                        className="css-1i325mk-Icon"
                        dangerouslySetInnerHTML={
                          {
                            "__html": {},
                          }
                        }
                      />
                    </div>
                    <div>
                      <div
                        className="css-15gyx2o-DonutChartWithLegend"
                      >
                        <div
                          className="css-qcf7xd-Text"
                        >
                          dashboard__progress_chart_legend
                        </div>
                        <div
                          className="css-1ggn5r6-Text"
                        >
                          24
                        </div>
                      </div>
                      <div
                        className="legend-entry css-1ya8xz5-DonutChartWithLegend"
                      >
                        <div
                          className="css-19o5m4d-DonutChartWithLegend"
                        />
                        <div
                          className="css-17zbm7y-Text"
                        >
                          dashboard__progress_chart_invited
                        </div>
                        <div
                          className="css-dlhgnj-Text"
                        >
                          6
                        </div>
                      </div>
                      <div
                        className="legend-entry css-1ya8xz5-DonutChartWithLegend"
                      >
                        <div
                          className="css-1e425ar-DonutChartWithLegend"
                        />
                        <div
                          className="css-17zbm7y-Text"
                        >
                          dashboard__progress_chart_active
                        </div>
                        <div
                          className="css-dlhgnj-Text"
                        >
                          6
                        </div>
                      </div>
                      <div
                        className="legend-entry css-1ya8xz5-DonutChartWithLegend"
                      >
                        <div
                          className="css-1noydjl-DonutChartWithLegend"
                        />
                        <div
                          className="css-17zbm7y-Text"
                        >
                          dashboard__progress_chart_completed
                        </div>
                        <div
                          className="css-dlhgnj-Text"
                        >
                          12
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="card-footer css-1decrb8-CardFooter"
                >
                  <div
                    className="css-17zbm7y-Text"
                  >
                    dashboard__progress_running
                  </div>
                  <div
                    className="css-l0jg45-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="css-18cybom-Card"
                onClick={[MockFunction]}
              >
                <div
                  className="card-header css-17xm4q9-CardHeader"
                >
                  <div
                    className="css-1cvfayx-Heading"
                  >
                    dashboard__progress_scoring_title
                  </div>
                  <div
                    className="css-1pmfd-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
                <div
                  className="card-content css-51sp4r-CardContent"
                >
                  <div
                    className="css-1xag63u-RatingProgressCard"
                  >
                    <div
                      className="donut-chart css-109bzso-DonutChartWithLegend"
                    >
                      <div
                        className="css-ow8vil-DonutChartWithLegend"
                      >
                        <svg
                          height="100%"
                          viewBox="0 0 100 100"
                          width="100%"
                        >
                          <path
                            d="M 50 95 A 45 45 0 1 1 50.00078539816336 94.9999999931461"
                            fill="none"
                            stroke="#d5d5d5"
                            strokeWidth={10}
                          />
                          <path
                            d=""
                            fill="none"
                            stroke="#b4b4b4"
                            strokeWidth={10}
                            style={{}}
                          >
                            <title>
                              rating__ongoing_count
                            </title>
                          </path>
                          <path
                            d=""
                            fill="none"
                            stroke="#4fcfc5"
                            strokeWidth={10}
                            style={{}}
                          >
                            <title>
                              rating__finished_count
                            </title>
                          </path>
                        </svg>
                        <div
                          className="css-1i325mk-Icon"
                          dangerouslySetInnerHTML={
                            {
                              "__html": {},
                            }
                          }
                        />
                      </div>
                      <div>
                        <div
                          className="css-15gyx2o-DonutChartWithLegend"
                        >
                          <div
                            className="css-qcf7xd-Text"
                          >
                            rating__raters_label
                          </div>
                          <div
                            className="css-1ggn5r6-Text"
                          >
                            0
                          </div>
                        </div>
                        <div
                          className="legend-entry css-jupuj5-DonutChartWithLegend"
                        >
                          <div
                            className="css-12vuskn-DonutChartWithLegend"
                          />
                          <div
                            className="css-17zbm7y-Text"
                          >
                            rating__ongoing_count
                          </div>
                          <div
                            className="css-dlhgnj-Text"
                          >
                            0
                          </div>
                        </div>
                        <div
                          className="legend-entry css-jupuj5-DonutChartWithLegend"
                        >
                          <div
                            className="css-1i6ffm0-DonutChartWithLegend"
                          />
                          <div
                            className="css-17zbm7y-Text"
                          >
                            rating__finished_count
                          </div>
                          <div
                            className="css-dlhgnj-Text"
                          >
                            0
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="card-footer css-1decrb8-CardFooter"
                >
                  <div
                    className="css-17zbm7y-Text"
                  >
                    dashboard__scoring_ongoing
                  </div>
                  <div
                    className="css-l0jg45-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="css-18cybom-Card"
                onClick={[MockFunction]}
              >
                <div
                  className="card-header css-1jjfdh1-CardHeader"
                >
                  <div
                    className="css-1cvfayx-Heading"
                  >
                    dashboard__progress_reports_title
                  </div>
                  <div
                    className="css-1pmfd-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
                <div
                  className="card-content css-51sp4r-CardContent"
                >
                  <div
                    className="css-wz4xt9-ReportingCard"
                  >
                    <div
                      className="css-17zbm7y-Text"
                    >
                      dashboard__progress_reports_finished_modules
                    </div>
                    <div
                      className="css-17zbm7y-Text"
                    >
                      0 / 1
                    </div>
                  </div>
                  <div
                    className="css-wz4xt9-ReportingCard"
                  >
                    <div
                      className="css-17zbm7y-Text"
                    >
                      dashboard__progress_reports_fully_rate_abbr
                    </div>
                    <div
                      className="css-17zbm7y-Text"
                    >
                      3
                       / 
                      6
                    </div>
                  </div>
                  <div
                    className="css-wz4xt9-ReportingCard"
                  >
                    <div
                      className="css-17zbm7y-Text"
                    >
                      dashboard__progress_reports_maximum_score
                    </div>
                    <div
                      className="css-17zbm7y-Text"
                    >
                      292
                    </div>
                  </div>
                  <div
                    className="css-wz4xt9-ReportingCard"
                  >
                    <div
                      className="css-qcf7xd-Text"
                    >
                      dashboard__progress_reports_survey_average
                    </div>
                    <div
                      className="css-qcf7xd-Text"
                    >
                      253
                    </div>
                  </div>
                  <div
                    className="css-ujroht-ProgressBar"
                  >
                    <div
                      className="progress-element css-1nvkhv2-ProgressBar"
                    />
                    <div
                      className="css-rsr7yg-ProgressBar"
                    />
                  </div>
                </div>
                <div
                  className="card-footer css-texso5-CardFooter"
                >
                  <div
                    className="css-17zbm7y-Text"
                  >
                    dashboard__progress_reports_missing_reports
                  </div>
                  <div
                    className="css-l0jg45-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="bottom-action-bar css-1koo7p0-ContentBottomActionbar"
  />
</div>
`;
